Assets/Scripts/DataTypes/Edges/Anchor.cs

```diff
@@ -0,0 +1,28 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using UnityEngine;
+
+namespace DataTypes
+{
+    public struct Anchor
+    {
+        public Road parentRoad;
+        public AnchorNumber number;
+        public Vector2 position;
+        public ImmutableArray<Lane> endingLanes;
+
+        public Anchor(AnchorNumber number, Vector2 position, IEnumerable<Lane> endingLanes)
+        {
+            parentRoad = null;
+            this.number = number;
+            this.position = position;
+            this.endingLanes = endingLanes.ToImmutableArray();
+        }
+
+        public Anchor(Road parentRoad, AnchorNumber number, Vector2 position, IEnumerable<Lane> endingLanes)
+            : this(number, position, endingLanes)
+        {
+            this.parentRoad = parentRoad;
+        }
```

Comment:
Constructor requires calling a Road-Object to assign as parentRoad, yet a Road-Object can not exist without already having two Anchors assigned directly. This would end up never getting called.

Reply:
8b59dd2